modules/dxp/apps/scim/scim-rest-impl/src/main/java/com/liferay/scim/rest/internal/manager/UserManagerImpl.java

```diff
@@ -581,6 +842,47 @@ private void _saveScimClientId(
 			portalUser.getUserId(), scimClientId);
 	}
 
+	private void _saveScimClientId(String scimClientId, UserGroup userGroup)
+		throws Exception {
+
+		ExpandoTable expandoTable = _expandoTableLocalService.fetchTable(
+			userGroup.getCompanyId(),
+			_classNameLocalService.getClassNameId(UserGroup.class.getName()),
+			ExpandoTableConstants.DEFAULT_TABLE_NAME);
+
+		if (expandoTable == null) {
+			expandoTable = _expandoTableLocalService.addTable(
+				userGroup.getCompanyId(), UserGroup.class.getName(),
+				ExpandoTableConstants.DEFAULT_TABLE_NAME);
+		}
+
+		ExpandoColumn expandoColumn = _expandoColumnLocalService.fetchColumn(
+			expandoTable.getTableId(), "scimClientId");
+
+		if (expandoColumn == null) {
+			expandoColumn = _expandoColumnLocalService.addColumn(
+				expandoTable.getTableId(), "scimClientId",
+				ExpandoColumnConstants.STRING);
+
+			UnicodeProperties unicodeProperties =
+				expandoColumn.getTypeSettingsProperties();
+
+			unicodeProperties.setProperty(
+				ExpandoColumnConstants.INDEX_TYPE,
+				String.valueOf(ExpandoColumnConstants.INDEX_TYPE_KEYWORD));
+
+			expandoColumn.setTypeSettingsProperties(unicodeProperties);
+
+			expandoColumn = _expandoColumnLocalService.updateExpandoColumn(
+				expandoColumn);
+		}
+
+		_expandoValueLocalService.addValue(
+			userGroup.getCompanyId(), UserGroup.class.getName(),
+			ExpandoTableConstants.DEFAULT_TABLE_NAME, expandoColumn.getName(),
+			userGroup.getPrimaryKey(), scimClientId);
+	}
+
```

Comment:
There is a lot of duplication with `_saveScimClientId(String scimClientId, com.liferay.portal.kernel.model.User portalUser)` here. Can we simplify/merge?